source/fuzz/repeated_pass_instances.h

```diff
@@ -0,0 +1,173 @@
+#include "source/fuzz/fuzzer_pass_permute_instructions.h"
+#include "source/fuzz/fuzzer_pass_propagate_instructions_up.h"
+#include "source/fuzz/fuzzer_pass_push_ids_through_variables.h"
+#include "source/fuzz/fuzzer_pass_replace_adds_subs_muls_with_carrying_extended.h"
+#include "source/fuzz/fuzzer_pass_replace_copy_memories_with_loads_stores.h"
+#include "source/fuzz/fuzzer_pass_replace_copy_objects_with_stores_loads.h"
+#include "source/fuzz/fuzzer_pass_replace_irrelevant_ids.h"
+#include "source/fuzz/fuzzer_pass_replace_linear_algebra_instructions.h"
+#include "source/fuzz/fuzzer_pass_replace_loads_stores_with_copy_memories.h"
+#include "source/fuzz/fuzzer_pass_replace_opphi_ids_from_dead_predecessors.h"
+#include "source/fuzz/fuzzer_pass_replace_opselects_with_conditional_branches.h"
+#include "source/fuzz/fuzzer_pass_replace_parameter_with_global.h"
+#include "source/fuzz/fuzzer_pass_replace_params_with_struct.h"
+#include "source/fuzz/fuzzer_pass_split_blocks.h"
+#include "source/fuzz/fuzzer_pass_swap_conditional_branch_operands.h"
+
+namespace spvtools {
+namespace fuzz {
+
+// This class has a distinct member for each repeated fuzzer pass (i.e., a
+// fuzzer pass that it makes sense to run multiple times).  If a member is null
+// then we do not have an instance of that fuzzer pass, i.e. it is disabled.
+// The class also provides access to the set of passes that are enabled.
+class RepeatedPassInstances {
+// This macro should be invoked below for every repeated fuzzer pass.  If a
+// repeated fuzzer pass is called FuzzerPassFoo then the macro invocation:
+//
+//    REPEATED_PASS_INSTANCE(Foo);
+//
+// should be used.  This adds a private member of type FuzzerPassFoo*, and
+// provides the following public methods:
+//
+// // Requires that SetPass has not been called previously with FuzzerPassFoo.
+// // Adds |pass| to the set of known pass instances.
+// void SetPass(std::unique_ptr<FuzzerPassFoo pass);
+//
+// // Returns a pointer to a pass instance of type FuzzerPassFoo that was
+// // previously registered via SetPass(), or nullptr if no such instance was
+// // registered
+// FuzzerPassFoo* GetFoo(); <-- Returns nullptr if no instance
+#define REPEATED_PASS_INSTANCE(NAME)                                     \
+ public:                                                                 \
+  FuzzerPass##NAME* Get##NAME() { return NAME##_; }                      \
+  void SetPass(std::unique_ptr<FuzzerPass##NAME> pass) {                 \
+    assert(NAME##_ == nullptr && "Attempt to set pass multiple times."); \
+    NAME##_ = pass.get();                                                \
+    passes_.push_back(std::move(pass));                                  \
+  }                                                                      \
+                                                                         \
+ private:                                                                \
+  FuzzerPass##NAME* NAME##_
```

Comment:
I think it would be better to explicitly initialize all fields here to `nullptr`. The current implementation can be safely used with list initialization (i.e. `RepeatedPassInstances s{};`) but this type of initialization is not very common in this project. This means that someone might accidentally ignore it, thus leaving the field uninitialized.

Reply:
Good point - done.

Reply:
BTW, I didn't notice it the first time but it appears you are missing a semicolon here.